### ISSUE TITLE
Fold evaluation program into pyserini directly

### ISSUE THREAD

**Issue by lintool** [castorini/pyserini#310]
From here: https://github.com/castorini/pyserini/blob/master/docs/pypi-replication.md

Currently, we can do this:
```
$ python -m pyserini.search --topics robust04 --index robust04 --output run.robust04.txt --bm25
```

But evaluation is still pretty annoying...

```
$ wget -O jtreceval-0.0.5-jar-with-dependencies.jar https://search.maven.org/remotecontent?filepath=uk/ac/gla/dcs/terrierteam/jtreceval/0.0.5/jtreceval-0.0.5-jar-with-dependencies.jar
$ wget https://raw.githubusercontent.com/castorini/anserini/master/src/main/resources/topics-and-qrels/qrels.robust04.txt
$ java -jar jtreceval-0.0.5-jar-with-dependencies.jar -m map -m P.30 qrels.robust04.txt run.robust04.txt
```

We still need to fetch the eval program; and `msmarco_doc_eval.py`/`msmarco_passage_eval.py` also...

Can use the pre-built index functionality to updownload the eval program into `~/.cache/pyserini/`?

Then we can implement something like:

```
$ python -m pyserini.eval.trec_eval ...
$ python -m pyserini.eval.msmarco_doc_eval ...
$ python -m pyserini.eval.msmarco_passage_eval ...
```

And each of these will just dispatch to the underlying programs...?

(Let's just get the eval program to work for now... I have ideas on dealing with the qrels next...)

Thoughts @MXueguang ?

**Comment by MXueguang** [comment]
looks good, should be easy to implement, will do.

**Comment by MXueguang** [comment]
when we implement `eval`,  say`python -m pyserini.eval.trec_eval`, we just wrap the bash command? i.e. run the `java` command in python script?

**Comment by lintool** [comment]
Yes, and then pass along their output.

**Comment by lintool** [comment]
Closed by #311
